packages/web-components/fast-element/src/templating/view.ts

```diff
@@ -154,6 +154,8 @@ export class HTMLView<TSource = any, TParent = any>
             let current = this.firstChild!;
             let next;
 
+            if (node.previousSibling === end) return;
```

Comment:
We could move the `const end` to be the first line in the `else` branch and then this check could be the second line, to return even a little bit earlier.